### DISCUSSION TITLE
There was an error updating. If you have uncommited changes in

### DISCUSSION THREAD

**Discussion by GronbergInfo** [ohmybash/oh-my-bash#643] [Q&A] [answered]
When i do a manual update (or try to) i get this and not sure what to do.

```
130 > upgrade_oh_my_bash
Updating Oh My Bash
error: cannot pull with rebase: You have unstaged changes.
error: please commit or stash them.
oh-my-bash: running 'git rebase --abort'...
fatal: No rebase in progress?
There was an error updating.
If you have uncommited changes in '/home/stefan/.oh-my-bash', please commit, stash or discard them and retry updating.
If you have your own local commits in '/home/stefan/.oh-my-bash' that conflict with the upstream changes, please resolve conflicts and merge the upstream manually.
[stefan@Hades10Gbe] [~/.oh-my-bash] ± master ↓4 U:358 ✗                                                                                                 [11:58:48]
1 >

```

**Comment by akinomyoga** [comment, marked as answer]
It is described in the error message:

"If you have uncommited changes in '/home/stefan/.oh-my-bash', please commit, stash or discard them and retry updating."

The contents of the oh-my-bash repository is managed by Git. If you directly modify the files of Oh My Bash, you need to tell what to do with that change to Git. There are options:

- You can register the change in the Git system by creating a *commit*. You can learn how to create a commit in web pages that explain the basics of Git. Basically, you can add the changes by the `git add ...` command. Then you can commit the added changes by the `git commit ...` command. Please refer to any pages introducing Git. In this case, if there is a conflicting change in the upstream repository, you'll need to resolve the conflict by yourself. The way to resolve conflicts are also described in many pages.
- Or you can *stash* the changes to temporarily revert your changes by running `git stash` in the Oh-My-Bash repository. The stashed changes can be later recovered by running the command `git stash pop` in the Oh-My-Bash repository.
- Another option is to discard the changes by running `git reset --hard` in the Oh-My-Bash repository. Please be careful before running this command. This command will remove your modifications to Oh My Bash in an unrecoverable way. If you mistakenly did it, you might still try to recover some of the files by `git fsck ...`, but it's not ensured.

**Comment by GronbergInfo** [comment]
Thanks, im unsure what changes i have made, if any (copied the omb folders from an old machine to a reinstalled one "only")
Stashed it and update worked.
